### PR TITLE
Use params from node '/**' from parameter YAML file

### PR DESCRIPTION
**Pull request by cottsay**
The short-term goal of this change is to enable the creation of a parameter YAML file which is applied to each node, regardless of node name or namespace.

Future work is to support all wildcard syntax in node names in parameter YAML files.

### CODE REVIEW

**Comment by cottsay** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7369/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3527/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6044/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7196/)

**Comment by cottsay** [comment]
I'm completely new to Python's C API, so if anything looks wonky, speak up.

File: rclpy/src/rclpy/_rclpy.c [hunk 1]
```diff
@@ -4523,21 +4523,28 @@ rclpy_get_node_parameters(PyObject * Py_UNUSED(self), PyObject * args)
...
-    Py_DECREF(params_by_node_name);
-    Py_DECREF(py_node_name_with_namespace);
-    return PyDict_New();
-  }
-  PyObject * node_params = PyDict_GetItem(params_by_node_name, py_node_name_with_namespace);
-  Py_DECREF(py_node_name_with_namespace);
-  if (NULL == node_params) {
-    Py_DECREF(params_by_node_name);
-    return NULL;
+  PyObject * node_params = PyDict_New();
```
Comment (jacobperron) [inline rclpy/src/rclpy/_rclpy.c]: If `PyDict_New()` fails it will return NULL, which should be guarded against.

File: rclpy/src/rclpy/_rclpy.c [hunk 2]
```diff
@@ -4523,21 +4523,28 @@ rclpy_get_node_parameters(PyObject * Py_UNUSED(self), PyObject * args)
...
+  PyObject * py_wildcard_name = PyUnicode_FromString("/**");
+
+  PyObject * current_key, * current_value;
+  Py_ssize_t current_index = 0;
+  while (PyDict_Next(params_by_node_name, &current_index, &current_key, &current_value)) {
+    // TODO(cottsay) implement further wildcard matching
+    if (PyObject_RichCompare(current_key, py_wildcard_name, Py_EQ) ||
```
Comment (jacobperron) [inline rclpy/src/rclpy/_rclpy.c]: Memory leak? Each call to [`PyObject_RichCompare`](https://docs.python.org/3/c-api/object.html#c.PyObject_RichCompare) returns a new reference that needs to be decref'd.

Comment (cottsay) [reply rclpy/src/rclpy/_rclpy.c]: Switched to [`PyObject_RichCompareBool`](https://docs.python.org/3/c-api/object.html#c.PyObject_RichCompareBool).

File: rclpy/src/rclpy/_rclpy.c (hunk 1 above)
Comment (cottsay) [reply rclpy/src/rclpy/_rclpy.c]: Fixed. Good catch.

**Pull request merged**